dima/react-movies/src/pages/mainpage/mainpage.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from "react";
+import ContentContainer from "../../components/contentContainer";
+import Header from "../../components/header";
+import SearchBar from "../../components/searchBar";
+import MoviesContainer from "../../components/moviesContainer";
+import ErrorBoundary from "../../components/errorBoundary";
+import MoviesResult from "../../components/moviesResult";
+import SortFilter from "../../components/sortFilter";
+import Footer from "../../components/footer";
+import QueryString from "query-string";
+import { useHistory, useLocation } from "react-router-dom";
+import SortProperty from "../../enums/SortProperty";
+import ParamsToPush from "../../interfaces/paramsToPush";
+
+import "./mainpage.css";
+import { useDispatch, useSelector } from "react-redux";
+import { resetMovies, fetchMovies } from "../../redux/moviesReducer";
+import { StateInterface } from "../../interfaces/stateInterface";
+import MoviesDataInterface from "../../interfaces/moviesDataInterface";
+
+const MainPage = () => {
+  const [currentSortType, setCurrentSortType] = useState(SortProperty.date);
+  let history = useHistory();
+  let location = useLocation();
```

Comment:
use **let** only when you gonna change a variable, in other cases always use **const**